lib/AstOfLlbc.ml

```diff
@@ -1025,57 +1026,68 @@ let rec expression_of_fn_ptr env depth (fn_ptr: C.fn_ptr) =
          anyhow. *)
       []
     else
-      List.concat_map (fun (trait_ref: C.trait_ref) ->
-        let name = string_of_name env (env.get_nth_trait_decl trait_ref.trait_decl_ref.trait_decl_id).name in
+      (* MUST have the same structure as build_trait_clause_mapping *)
```

Comment:
this populates the trait method references at call-site, so we need to materialize function pointers exactly as expected by the signature, which is determined by `build_trait_clause_mapping`

Reply:
aybe you could use something more structured than a list then, e.g. a tree that follows the parent relationships